mrjob/py2.py

```diff
@@ -0,0 +1,170 @@
+"""Minimal utilities to make Python work for 2.6+ and 3.3+
+
+Strategies for making `mrjob` work across Python versions:
+
+Bytes vs. Unicode
+-----------------
+
+It's tempting to use `from __future__ import unicode_literals` and require
+that all non-byte strings be unicode. But that doesn't really make sense for
+Python 2, where str (bytes) and unicode can be used interchangeably.
+
+So really our string datatypes fall into two categories, bytes, and
+"strings", which are either ``str`` (i.e., bytes) or ``unicode`` in Python 2,
+and ``str`` (i.e. unicode) in Python 3.
+
+These things should always be bytes:
+
+- input data files
```

Comment:
That last sentence is wayy too complex.

It might be best to discuss in person.

I take issue with "Python 2, where str (bytes) and unicode can be used interchangeably". This is *only* true for English strings.